### ISSUE TITLE
The importance of epochs when training yolo

### ISSUE THREAD

**Issue by ggior** [AlexeyAB/darknet#5508]
I am wondering why the number of images has no influence on the number of iterations when training. Here is an example to to make my question clearer:
Suppose we have 6400 images for a training to recognize 4 classes. Based on AlexeyAB explanations, we keep batch= 64, subdivisions = 16 and write max_batches = 8000 since max_batches is determined by #classes x 2000. 

Since we have 6400 images, a complete epoch requires 100 iterations.  Therefore this training ends after 80 epochs.
Now, suppose that we have 12800 images. In that case, an epoch needs 200 iterations. Therefore the training ends after 40 epochs.
Since an epoch refers to one cycle through the full training dataset, I'm wondering why we don't increase the number of iterations when our dataset increases, in order to keep the number of epochs constant.

**Comment by AlexeyAB** [comment]
In general it should be  `max_batches = max(6000, max(2000*classes, number_of_images) )`

* if you are an experienced user, then you will train either while reducing loss or while accuracy is growing, **even without any manual**

* while novice user usually train 50 iterations by using 10 images and complain that there are no results, but they should train at least 6000 iterations

----

I just slightly fixed readme:

https://github.com/AlexeyAB/darknet#how-to-train-to-detect-your-custom-objects

change line max_batches to (classes*2000 but not less than number of training images, and not less than 6000), f.e. max_batches=6000 if you train for 3 classes

https://github.com/AlexeyAB/darknet#when-should-i-stop-training

Usually sufficient 2000 iterations for each class(object), but not less than number of training images and not less than 6000 iterations in total. But for a more precise definition when you should stop training, use the following manual:

**Comment by ggior** [comment]
Hi @AlexeyAB, thank you very much for your answer.  

I would consider myself a novice user but I have been training with ~9000 images and I've been using the flag -map in order to know when to stop the training. But you are right, I did put too much emphasis on the formula for the number of iterations instead of focusing entirely on the mAP.

Thanks again for your help.
